apps/web/src/pages/api/notifications/webhook.ts

```diff
@@ -0,0 +1,85 @@
+import { eq } from "drizzle-orm";
+import { drizzle } from "drizzle-orm/node-postgres";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { Pool } from "pg";
+import { gmUsers } from "db/schema";
+
+const dbConnectionString = process.env.DB_POOLING_URL;
+
+if (!dbConnectionString) {
+  throw new ReferenceError(
+    "process.env.DB_POOLING_URL is missing in environment variables"
+  );
+}
+
+const pool = new Pool({
+  connectionString: dbConnectionString,
+});
+
+const db = drizzle(pool);
+
+// eslint-disable-next-line consistent-return
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    // Method Not Allowed
+    res.status(405).json({ success: false });
+  }
+
+  const { id, event, account, dappUrl } = req.body;
+
+  console.log(
+    `[API] Webhook ID ${id} received event "${event}" for account ${account} on dapp ${dappUrl}`
+  );
+  console.log({ id, event, account, dappUrl, reqHeaders: req.headers });
+
+  if (!account || !event) {
+    return res.status(400).json({ success: false });
+  }
+
+  if (event === "subscribed") {
+    const existingUser = await db
+      .select()
+      .from(gmUsers)
+      .where(eq(gmUsers.account, account));
+
+    if (existingUser.length > 0) {
+      return res.status(200).json({ success: true });
+    }
+
+    try {
+      await db.insert(gmUsers).values({
+        account,
+      });
+      return res.status(200).json({ success: true });
+    } catch (error) {
+      console.error(`Subscribing account ${account} failed: `, error);
+      // if (error instanceof Error) {
+        return res
+          .status(500)
+          .json({ success: false, message: error });
+      // }
+    }
+  } else if (event === "unsubscribed") {
+    try {
+      await db.delete(gmUsers).where(eq(gmUsers.account, account));
+      return res.status(200).json({ success: true });
+    } catch (error) {
+      console.error(`Unsubscribing account ${account} failed: `, error);
```

Comment:
intrntionallly pushed without clean. playing around with different methods of creating custom extened wc connector that supports push client. theres a few ways ive beeen trying which is why theres a lot of duplicated/commented code

1) extend wagmi create custom connector
2) write custom ethers provider (universal provider) using sign-client combined with push (requires implement own json rpc spec, so this would make wagmi redundant)
3) use current wagmi and then implement the pushclient in an extended context provider, ...ok but requires second sig

Reply:
also lot of new packages installed. tbd which we actually need